pages/account/Dashboard.js

```diff
@@ -1,43 +1,113 @@
 // friends
 // Pull in timezones from a json file and display them:
 // import
-import styles from '../../styles/dashboard.module.css'
+import styles from '../../styles/dashboard.module.css';
 import AddFriend from '../../components/AddFriend';
+import {useState, useEffect} from 'react';
 
-const UTC_OFFSETS = require('/data/timezones.json');
-const timeSettings = { hour: '2-digit', minute: '2-digit' };
+const UTC_OFFSETS = require('/data/tznames.json');
+const timeSettings = {hour: '2-digit', minute: '2-digit'};
 
 function DisplayOffsets(props) {
-    const offsets = props.offsets
-    // Display offsets
-    return (
-        offsets.map((offset) =>
-            <div className={styles.zones} key={offset}>
-                {offset}
-                <hr className={styles.line_break}></hr>
-                {/*TODO: Check this against friend functionalities not magic number +02:00*/}
-                {(offset === '+02:00') ? new Date().toLocaleTimeString([], { ...timeSettings }) : ''}
-            </div>
-        )
-    );
+  const offsets = props.offsets;
+  const friendList = props.friendList;
+  // Create zones of friends to check against
+  const zones = [];
+  friendList.map((friend) => zones.push(friend.timezone));
+  // Display offsets
+  return (
+    offsets.map((offset) =>
+      <div className={styles.zones} key={offset.utc_offset}>
+        {offset.utc_offset}
+        <hr className={styles.line_break}></hr>
+        {(zones.includes(offset.utc_offset)) ? new Date().toLocaleTimeString([],
+            {timeZone: (offset.timezone), ...timeSettings}) : ''}
+      </div>
+    )
+  );
+}
+
+function DisplayFriends(props) {
```

Comment:
I would separate DisplayFriends into its own component and import to Dashboard

Reply:
Got that and DisplayOffsets as well!